packages/TypeDeclaration/tests/Rector/ClassMethod/AddArrayReturnDocTypeRector/Fixture/skip_mixed_array.php.inc

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Rector\TypeDeclaration\Tests\Rector\ClassMethod\AddArrayReturnDocTypeRector\Fixture;
+
+final class SkipMixedArray
+{
+    /**
+     * @return mixed[]
+     */
+    public function someMethod(): array
```

Comment:
The more I see these issue, the more I think the mixed type should be configurable. Explicit conversion by default (*my* approach),
optionally preffered (*your* approach).

What do you think?

Reply:
I guess the question is : do people often use Rector to upgrade code which has many "mixed" type already written to get more specific types (guessed from the code itself) ? If yes, then it makes sense that Rector modifies existing mixed phpdocs. Otherwise, I would say that Rector shouldn't touch an existing mixed type (letting the developer be in control) and add specific types when the phpdoc is missing.